packages/superinput/src/Superinput.tsx

```diff
@@ -0,0 +1,108 @@
+import { completionKeymap } from '@codemirror/autocomplete'
+import { closeBrackets, closeBracketsKeymap } from '@codemirror/closebrackets'
+import { classHighlightStyle } from '@codemirror/highlight'
+import { history, historyKeymap } from '@codemirror/history'
+import { javascript } from '@codemirror/lang-javascript'
+import { EditorState, Extension } from '@codemirror/state'
+import { placeholder as placeholderExt, keymap, EditorView, ViewUpdate } from '@codemirror/view'
+import React, { useEffect, useState, useRef, MutableRefObject } from 'react'
+
+import EditorFrame from './EditorFrame'
+import EvalPanel from './EvalPanel'
+import { bpAutocomplete, BPLang, hoverInspect, exprDecorator } from './extensions'
+import { SI_TYPES, ISiProps } from './types'
+import { isError, evalStrTempl } from './utils/tokenEval'
+
+export default function Superinput({
+  value,
+  globs,
+  onChange,
+  placeholder,
+  autoFocus = false,
+  type = SI_TYPES.TEMPLATE,
+  noGlobsEvalMsg = ''
+}: ISiProps) {
+  const editor = useRef() as MutableRefObject<HTMLInputElement>
+  const [panel, setPanel] = useState('')
+  const [view, setView] = useState() as [EditorView, React.Dispatch<React.SetStateAction<EditorView>>]
+
+  useEffect(() => {
+    const onUpdate = (update: ViewUpdate) => {
+      const { focusChanged, docChanged, view } = update
+      const value = update.state.doc.sliceString(0)
+
+      if (!view.hasFocus) {
+        setPanel('')
+      } else if (focusChanged || docChanged) {
+        setPanel(!globs ? noGlobsEvalMsg : evalStrTempl(value, globs) || '')
+      }
+
+      if (docChanged && onChange) {
+        onChange(value)
+      }
+    }
+
+    let typeExt: Extension[] = []
+    const keymapList = [...closeBracketsKeymap, ...historyKeymap]
+
+    if (type === SI_TYPES.TEMPLATE) {
+      typeExt = [BPLang(), exprDecorator(globs)]
+      keymapList.push(...completionKeymap)
+    } else if (type === SI_TYPES.EXPRESSION || type === SI_TYPES.BOOL) {
+      typeExt = [javascript()]
+    }
+
+    const extensions = [
+      EditorView.updateListener.of(onUpdate),
+      EditorView.lineWrapping,
+      classHighlightStyle,
+      placeholderExt(placeholder || ''),
+      ...typeExt,
+      hoverInspect(globs),
+      bpAutocomplete(globs),
+      history(),
+      closeBrackets(),
+      keymap.of(keymapList)
+    ]
+
+    const state = EditorState.create({
+      doc: value,
+      extensions
+    })
+    const newView = new EditorView({ state, parent: editor.current })
+    setView(newView)
+
+    if (autoFocus) {
+      newView.focus()
+    }
+
+    return () => {
+      if (view) {
+        view.destroy()
+      }
```

Comment:
just a detail, with optional chaining you can simply use `view?.destroy()` instead of checking if it is not null before